content/tutorials/permissionless-paymaster/10.index.md

```diff
@@ -0,0 +1,439 @@
+    // Get the maxNonce allowed to user. Here we ensure it's currentNonce.
+    const maxNonce = BigNumber.from(
+      await provider.getTransactionCount(account.address || "") 
+ );
+    // You can also check for min Nonce from the NonceHolder System contract to fully ensure as ZKsync support arbitrary nonce.
+    const nonceHolderAddress = "0x0000000000000000000000000000000000008003";
+    const nonceHolderAbi = [
+      "function getMinNonce(address _address) external view returns (uint256)",
+ ];
+    const nonceHolderContract = new Contract(
+      nonceHolderAddress,
+      nonceHolderAbi,
+      provider
+ );
+    const maxNonce2 = await nonceHolderContract.callStatic.getMinNonce(
+      account.address || ""
+ );
+    console.log(maxNonce2.toString());
+    // -----------------
+    // Get the expiration time. Here signature will be valid up to 120 sec.
+    const currentTimestamp = BigNumber.from(
+ (await provider.getBlock("latest")).timestamp
+ );
+    const expirationTime = currentTimestamp.add(120);
+    // Get the current gas price.
+    const maxFeePerGas = await provider.getGasPrice();
+    // Add paymaster overhead gas to be on the safe side.
+    const gasLimit = estimateGas.add(65000);
+    // ------------------------------------------------------------------------------------
+    const [paymasterAddress, signature, signerAddress] = await getSignature(
+      account.address.toString(),
+      daiContractConfig.address.toString(),
+      expirationTime,
+      maxNonce,
+      maxFeePerGas,
+      gasLimit
+ );
+    console.log("Signer: " + signerAddress);
+    // We encode the extra data to be sent to paymaster
+    // Notice how it's not required to provide from, to, maxFeePerGas, and gasLimit as per the signature above.
+    // That's because paymaster will get it from the transaction struct directly to ensure it's the correct user.
+    const innerInput = ethers.utils.arrayify(
+      abiCoder.encode(
+ ["uint256", "uint256", "address", "bytes"],
+ [
+          expirationTime, // As used in the above signature
+          maxNonce, // As used in the above signature
+          signerAddress, // The signer address
+          signature,
+ ]
+ ) // Signature created in the above snippet. get from API server
+ );
+    // getPaymasterParams function is available in zksync-ethers
+    const paymasterParams = utils.getPaymasterParams(
+      paymasterAddress, // Paymaster address
+ {
+        type: "General",
+        innerInput: innerInput,
+ }
+ );
+    // Returns paymaster params, gas fee, gas limit
+    return [paymasterParams, maxFeePerGas, gasLimit];
+ };
+```
+
+##### **5.2** Estimate gas and call the above function
+
+```javascript [src/components/WriteContract.tsx]
+  // Estimate gas
+    const estimateGas = await contract.estimateGas.approve(spender,amount);
+    const [paymasterParams, maxFeePerGas, gasLimit] = await preparePaymasterParam(account, estimateGas);
```

Comment:
it seems like this line of code is also needed for the `account` variable: `const { account, getSigner, getProvider, network } = useEthereum();`

Reply:
Added more description and comments for this step. Since, we are adding in the file, those variables are already defined in the template.